pallets/collator-selection/src/lib.rs

```diff
@@ -305,6 +320,9 @@ pub mod pallet {
 				);
 			}
 
+			// Invulnerables must be sorted for removal.
+			bounded_invulnerables.sort();
```

Comment:
We could also alternatively expect the caller to provide a sorted vector `O(logn)` and verify on-chain that vector is sorted `O(n)`.

Reply:
Yeah, I have considered this. I decided against it because calling this function requires governance, so (a) it's not very accessible to users, and (b) a mistake in sorting means another trip through a referendum track, which can take 28 days on Polkadot. And the impact is small, because we generally expect to be dealing with very short vecs, almost certainly with fewer than 20 elements (probably <10).

I can add that to the function docs re expected usage, and if someone wants to use it for larger sets then we can make a change.

Reply:
Sounds fair.